octobot_evaluators/evaluator/strategy_evaluator.py

```diff
@@ -57,6 +141,225 @@ async def matrix_callback(self,
+            return True
+        return False
+
+    def _get_available_node_paths(self,
+                                  matrix_id,
+                                  evaluator_type,
+                                  exchange_name,
+                                  cryptocurrency,
+                                  symbol,
+                                  use_cache=True):
+        if use_cache:
+            try:
+                return self.available_node_paths_cache[matrix_id][exchange_name][evaluator_type]
+            except KeyError:
+                # No cache usage here to be use to refresh data
+                node_paths = self._inner_get_available_node_paths(matrix_id, evaluator_type, exchange_name,
+                                                                  cryptocurrency, symbol, use_cache=False)
+                if matrix_id not in self.available_node_paths_cache:
+                    self.available_node_paths_cache[matrix_id] = {}
+                if exchange_name not in self.available_node_paths_cache[matrix_id]:
+                    self.available_node_paths_cache[matrix_id][exchange_name] = {
+                        evaluator_type: node_paths
+                    }
+                return node_paths
+        return self._inner_get_available_node_paths(matrix_id, evaluator_type, exchange_name,
+                                                    cryptocurrency, symbol, use_cache=use_cache)
+
+    def _inner_get_available_node_paths(self, matrix_id, evaluator_type, exchange_name, cryptocurrency, symbol,
+                                        use_cache=True):
+        available_time_frames = self.get_available_time_frames(matrix_id, exchange_name, evaluator_type,
+                                                               cryptocurrency, symbol, use_cache=use_cache)
+        to_check_time_frames = [tf
+                                for tf in available_time_frames
+                                if TimeFrames(tf) in self.strategy_time_frames]
+        available_evaluators = self._get_available_evaluators(matrix_id, exchange_name, evaluator_type,
+                                                              use_cache=use_cache)
+        return [
+            get_matrix_default_value_path(tentacle_name=evaluator,
+                                          tentacle_type=evaluator_type,
+                                          exchange_name=exchange_name,
+                                          cryptocurrency=cryptocurrency,
+                                          symbol=symbol,
+                                          time_frame=time_frame)
+            for time_frame in to_check_time_frames
+            for evaluator in available_evaluators
+        ]
+
+    def _save_last_evaluation(self, matrix_id, exchange_name, evaluator_type, tentacle_name,
+                              cryptocurrency, symbol, time_frame):
+        update_time = get_tentacle_eval_time(matrix_id,
```

Comment:
Can we merge this line with the next one ? Or it'll be too difficult to read ?

Reply:
I think we should keep it like this

Reply:
Why ? There is absolutely no difference if you add a break line after `time_frame` argument.